app/models/comfy/cms/page.rb

```diff
@@ -90,14 +90,7 @@ def ever_been_published?
   def publishable?
     keywords.all?(&:publishable?)
   end
-
-  def published_at
-    if published?
-      updated_at
-    elsif scheduled?
-      scheduled_on
-    elsif ever_been_published?
-      revisions.detect { |revision| revision.data[:event] == 'published' }.created_at
-    end
-  end
 end
+
+Comfy::Cms::Page.state_machine.events[:publish].timestamp = :published_at
+Comfy::Cms::Page.state_machine.events[:publish_changes].timestamp = :published_at
```

Comment:
So when an article is published we set the published_at field using the state machine. Instead of doing the code which attempting to dynamically work out the published_at value